opts/port.go

```diff
@@ -148,20 +148,31 @@ func ConvertPortToPortConfig(
 	ports := []swarm.PortConfig{}
 
 	for _, binding := range portBindings[port] {
+		hostPorts := []uint64{}
```

Comment:
Thank you for your PR!
I think this code can be simplified a little (and avoid the double iteration):
```go
for _, binding := range portBindings[port] {
    if binding.HostIP != "" && binding.HostIP != "0.0.0.0" {
        logrus.Warnf("ignoring IP-address (%s:%s:%s) service will listen on '0.0.0.0'", binding.HostIP, binding.HostPort, port)
    }
    startHostPort, endHostPort, err := nat.ParsePortRange(binding.HostPort)
    if err != nil && binding.HostPort != "" {
        return nil, fmt.Errorf("invalid hostport binding (%s) for port (%s)", binding.HostPort, port.Port())
    }
    for i := startHostPort; i <= endHostPort; i++ {
        ports = append(ports, swarm.PortConfig{
            //TODO Name: ?
            Protocol:      swarm.PortConfigProtocol(strings.ToLower(port.Proto())),
            TargetPort:    uint32(port.Int()),
            PublishedPort: uint32(hostPort),
            PublishMode:   swarm.PortConfigPublishModeIngress,
        })
    }
}
```
Does it sounds good to you?
And is it possible to add some tests on this feature in opts/port_test.go?